### ISSUE TITLE
Converting between camel-case enums and the Google API equivalents is a pain

### ISSUE THREAD

**Issue by richardthombs** [ericnewton76/gmaps-api-net#84]
Our API uses camel-case for enum values, ie `AdministrativeAreaLevel1` whereas the Google Maps API uses lowercase with words separated by underscores: `administrative_area_level_1`.

There is too much code that does nothing magical apart from being a massive switch statements that convert strings to enums and back again. This makes it harder to keep the code up to date with the API, because adding and removing enum values means code has to be updated in several locations. It also leads to bugs like #74.

I suggest we either:

1. Replace all our camel-case enums with ones that match the Google API so we can do a direct mapping using `Enum.Parse` and `ToString`. Obviously this would break existing code, although nothing that couldn't be fixed with minimal effort.

1. We create an enum helper that can do the conversions for us. This would involve iterating through the characters of an enum name in order to find the uppercase characters and adding an underscore, so it is likely to incur a performance hit.

Personally I favour option 1 as it will be faster and simpler.

**Comment by ericnewton76** [comment]
Doesnt the Newtonsoft Json parser library have an option similar to this?  Like CamelCaseProperty or something?   I'm going to look and update this comment after

**Comment by richardthombs** [comment]
So long as the conversions are handled automatically, rather than relying on a human to correctly convert "place_of_worship" -> "PlaceOfWorship" and back again, I'm all for it.

I'm a big fan of optimise later :smile:

And this way we'll keep the API the same and after all, CamelCase is probably what most of our users expect.

As you might have guessed from the flood of pull requests, I'm between jobs at the moment, so I'll look into implementing this, if you like?

**Comment by ericnewton76** [comment]
Between jobs: Lol, as I am, too...

**Comment by richardthombs** [comment]
Well maybe I can pay you to make this change?

If not, I just found this: http://www.newtonsoft.com/json/help/html/NamingStrategySnakeCase.htm

So maybe it's built into newer versions of Newtonsoft.Json.

**Comment by ericnewton76** [comment]
Yes I believe its in the newer Jsonsoft's.  I was going to try to tackle this.

**Comment by richardthombs** [comment]
If we wanted to avoid requiring v10 of Newtonsoft.Json, then I'm sure we can implement this as a custom converter in older versions of Newtonsoft.Json without too much trouble.

But I don't see that much a problem in requiring folks to upgrade to v10.
